api/v1alpha1/capiprovider_types.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright SUSE 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "sigs.k8s.io/cluster-api-operator/api/v1alpha2"
+)
+
+const (
+	// ProviderFinalizer is the finalizer apply on the CAPI Provider resource.
+	ProviderFinalizer = "capiprovider.turtles.cattle.io"
+)
+
+// CAPIProviderSpec defines the desired state of CAPIProvider.
+// +kubebuilder:validation:XValidation:message="CAPI Provider version should be in the semver format",rule="!has(self.version) || self.version.matches(r\"\"\"^([0-9]+)\\.([0-9]+)\\.([0-9]+)(?:-([0-9A-Za-z-]+(?:\\.[0-9A-Za-z-]+)*))?(?:\\+[0-9A-Za-z-]+)?$\"\"\")"
+//
+//nolint:lll
+type CAPIProviderSpec struct {
+	// Name is the name of the provider to enable
+	// +required
+	// +kubebuilder:validation:Enum=aws;azure;gcp;docker;rke2
+	Name string `json:"name"`
+
+	// Type is the type of the provider to enable
+	// +required
+	// +kubebuilder:validation:Enum=infrastructure;core;controlPlane;bootstrap;addon
+	Type string `json:"type"`
```

Comment:
We can create a type alias/constants for the values.

Reply:
Done